### PR TITLE
[INFRA] modularisation - directory file management

### PR DESCRIPTION
**Pull request by Irallia**
Resolves https://github.com/seqan/iGenVar/issues/66

Part of Epic https://github.com/seqan/iGenVar/issues/44.

Was blocked by https://github.com/seqan/igenvar/issues/42

### CODE REVIEW

File: include/modules/sv_detecting_methods/analyze_cigar_method.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * \param junctions         vector for storing junctions
+ * \param insertions        vector for storing insertion_alleles
+ * \param min_length        minimum length of variants to detect (default 30 bp)
+ * \param insertion_file    output file for insertion alleles
+ *
+ * \details This function steps through the CIGAR string and stores junctions with their position in reference and read.
+ *          We distinguish 4 cases of CIGAR operation characters:
+ *          1. M, =, X: For matches and mismatches (Alignment column containing two letters. This could contain two
+ *                      different letters (mismatch) or two identical letters), we step through ref and read.
+ *          2. I:       For insertons (gap in the reference sequence) -> Insertions cause two junctions ( (1) from the
```
Comment (simonsasse) [inline include/modules/sv_detecting_methods/analyze_cigar_method.hpp:19]:
```suggestion
 *          2. I:       For insertions (gap in the reference sequence) -> Insertions cause two junctions ( (1) from the
```

File: include/modules/sv_detecting_methods/analyze_sa_tag_method.hpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * \param delim delimiter
+ */
+template <class Container>
+void split_string(const std::string& str, Container& cont, char delim = ' ');
+
+/*! \brief Parse the SA tag from the SAM/BAM alignment of a chimeric/split-aligned read. Build
+ *         [aligned_segments](\ref AlignedSegment), one for each alignment segment of the read.
+ *
+ * \param sa_string         "SA" tag string
+ * \param aligned_segments  vector of [aligned_segments](\ref AlignedSegment).
```
Comment (simonsasse) [inline include/modules/sv_detecting_methods/analyze_sa_tag_method.hpp:18]: There are no full stops after any other param so I assume there shouldn't be one here ? :)
```suggestion
 * \param aligned_segments  vector of [aligned_segments](\ref AlignedSegment)
```

File: include/modules/sv_detecting_methods/analyze_sa_tag_method.hpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ *          We add all segments to our candidate list `aligned_segments` and examine them in the following function
+ *          `analyze_aligned_segments()`.
+ *          For more information about this tag, see the
+ *          ([Map Optional Fields Specification](https://github.com/samtools/hts-specs/blob/master/SAMtags.pdf)).
+ */
+void retrieve_aligned_segments(std::string sa_string, std::vector<AlignedSegment> & aligned_segments);
+
+/*! \brief Build junctions out of aligned_segments.
+ *
+ * \param aligned_segments  vector of [aligned_segments](\ref AlignedSegment).
```
Comment (simonsasse) [inline include/modules/sv_detecting_methods/analyze_sa_tag_method.hpp:33]: Maybe full stops make sense after references in params ? I'll mark it anyways :) 
```suggestion
 * \param aligned_segments  vector of [aligned_segments](\ref AlignedSegment)
```

File: src/modules/sv_detecting_methods/analyze_cigar_method.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        using seqan3::get;
+        int32_t length = get<0>(pair);
+        seqan3::cigar_op operation = get<1>(pair);
+        if (operation == 'M'_cigar_op || operation == '='_cigar_op || operation == 'X'_cigar_op)
+        {
+            pos_ref += length;
+            pos_read += length;
+        }
+        else if (operation == 'I'_cigar_op) // I: Insertion (gap in the reference sequence).
```
Comment (simonsasse) [inline src/modules/sv_detecting_methods/analyze_cigar_method.cpp:39]: dot or not ? 
```suggestion
        else if (operation == 'I'_cigar_op) // I: Insertion (gap in the reference sequence)
```

**Review by simonsasse — CHANGES_REQUESTED** [review]
Tiny stuff. Your code is really well documented and understandable !
The only inconsistent thing I found is how you use full stops after comments. :)

**Review by simonsasse — APPROVED** [review]
Looks good :)

File: include/modules/clustering/simple_clustering_method.hpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include "structures/cluster.hpp"   // for class Cluster
+
+/*! \brief This method compares junctions, and if they are equal, one will be removed from the junction list.
+ *
+ * \param junctions A vector of junctions
+ */
```
Comment (eldariont) [inline include/modules/clustering/simple_clustering_method.hpp:8]: I would rephrase a bit:

`This method clusters junctions by merging identical junctions into a single cluster object.`

File: src/detect_breakends/junction_detection.cpp [hunk 6]
```diff
@@ -102,19 +100,7 @@ void detect_junctions_in_alignment_file(const std::filesystem::path & alignment_
...
-                            {
-                                std::vector<AlignedSegment> aligned_segments{};
-                                auto strand = (hasFlagReverseComplement(flag) ? strand::reverse : strand::forward);
-                                aligned_segments.push_back(AlignedSegment{strand, ref_name, pos, mapq, cigar});
-                                retrieve_aligned_segments(sa_tag, aligned_segments);
-                                std::sort(aligned_segments.begin(), aligned_segments.end());
-                                analyze_aligned_segments(aligned_segments, junctions, query_name);
-                            }
-                        }
+                        analyze_sa_tag(query_name, flag, ref_name, pos, mapq, cigar, tags, junctions);
```
Comment (eldariont) [inline src/detect_breakends/junction_detection.cpp:103]: I would leave the check of the supplementary flag and the check whether the SA tag is empty out of the method:

```
                    case 2: // Detect junctions from split read evidence (SA tag, primary alignments only)
                        if (!hasFlagSupplementary(flag))
                        {
                            std::string sa_tag = tags.get<"SA"_tag>();
                            if (!sa_tag.empty())
                            {
                                analyze_sa_tag(query_name, flag, ref_name, pos, mapq, cigar, tags, junctions);
                            }
                        }
                        break;
```

This makes it clearer what is actually happening without looking into the method. To me, that feels much more natural and makes sure that the method carries out only the well-defined task of analyzing the SA tag.

File: test/api/junction_detection_test.cpp [hunk 7]
```diff
@@ -32,7 +32,6 @@ TEST(junction_detection, fasta_out_not_empty)
         "Reference\tchr21\t41972616\tForward\tRead\t0\t2294\tForward\t1\n"
         "Reference\tchr21\t41972616\tReverse\tRead\t0\t3975\tReverse\t1\n"
         "Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\t1\n"
-        "Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\t2\n"
```
Comment (eldariont) [inline test/api/junction_detection_test.cpp:35]: Do you understand why this line is removed from all of the test cases? I would not expect a change in output based on your changes.

**Review by eldariont — CHANGES_REQUESTED** [review]
Thanks for this great work 👍
I discovered a few issues that should be quick to fix.

File: test/api/junction_detection_test.cpp (hunk 7 above)
Comment (Irallia) [reply test/api/junction_detection_test.cpp]: I don't know. I was confused aswell. But after my rebase, this wierd change disapeard again. Must be a mistake I made, which got removed now. Good that we have tests :)

**Review by eldariont — APPROVED** [review]
